sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -420,22 +418,13 @@ bool _parse_options(int msg_type, gnrc_rpl_instance_t *inst, gnrc_rpl_opt_t *opt
                     first_target = target;
                 }
 
-                uint32_t fib_dst_flags = 0;
-
-                if (target->prefix_length <= IPV6_ADDR_BIT_LEN) {
-                    fib_dst_flags = ((uint32_t)(target->prefix_length) << FIB_FLAG_NET_PREFIX_SHIFT);
-                }
-
-                DEBUG("RPL: adding fib entry %s/%d 0x%" PRIx32 "\n",
+                DEBUG("RPL: adding FT entry %s/%d 0x%" PRIx32 "\n",
                       ipv6_addr_to_str(addr_str, &(target->target), sizeof(addr_str)),
-                      target->prefix_length,
-                      fib_dst_flags);
-
-                fib_add_entry(&gnrc_ipv6_fib_table, dodag->iface, target->target.u8,
-                              sizeof(ipv6_addr_t), fib_dst_flags, src->u8,
-                              sizeof(ipv6_addr_t), FIB_FLAG_RPL_ROUTE,
-                              (dodag->default_lifetime * dodag->lifetime_unit) *
-                              MS_PER_SEC);
+                      target->prefix_length);
+
+                gnrc_ipv6_nib_ft_add(&(target->target), target->prefix_length, src,
+                                     dodag->iface,
+                                     dodag->default_lifetime * dodag->lifetime_unit);
```

Comment:
Not part of this PR, but is it really necessary to store the `lifetime_unit` of a DODAG Oo?

Reply:
The lifetime and the lifetimeunit are both distributed in DIOs. Need to store that to further distribute it (:

Reply:
Ok! :-)